src/java.base/share/classes/jdk/internal/foreign/ArenaImpl.java

```diff
@@ -29,8 +29,17 @@
 import java.lang.foreign.MemorySegment;
 import java.lang.foreign.MemorySegment.Scope;
 
+import jdk.internal.misc.Unsafe;
+
 public final class ArenaImpl implements Arena {
 
+    static {
+        // initialize MemorySegment before touching NativeMemorySegmentImpl to avoid deadlock
+        // if multiple threads try to initialize NativeMemorySegmentImpl and MS through allocateNoInit
+        // and a method on MS at the same time
+        Unsafe.getUnsafe().ensureClassInitialized(MemorySegment.class);
```

Comment:
If you want to avoid unsafe, another way is to use:
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#ensureInitialized(java.lang.Class)

Reply:
Yeah. I figured using Unsafe is a little more lightweight, especially for FileChannel::map. Maybe it doesn't matter that much though.

Would you prefer using Methodhandles::ensureInitialized?